Sources/Testing/Test+Discovery.swift

```diff
@@ -100,8 +101,9 @@ extension Test {
       if useLegacyMode && result.isEmpty {
         let generators = Generator.allTypeMetadataBasedTestContentRecords().lazy.compactMap { $0.load() }
         await withTaskGroup { taskGroup in
-          for generator in generators {
-            taskGroup.addTask { await generator.rawValue() }
+          for (i, generator) in generators.enumerated() {
+            let taskName = "legacy test discovery (#\(i))"
```

Comment:
Although our implementation refers to this as "legacy", it's still the current supported discovery mechanism in production toolchains. It might be better for the user-visible task name to omit the word "legacy" until we enable using the new mechanism in production. (We could leave a "FIXME" comment to remind us to update this when that happens.)

Reply:
Sure